tooling/lsp/src/requests/hover/from_visitor.rs

```diff
@@ -0,0 +1,78 @@
+use std::str::FromStr;
+
+use acvm::FieldElement;
+use fm::{FileId, FileMap};
+use lsp_types::{Hover, HoverContents, MarkupContent, MarkupKind, Position};
+use noirc_errors::{Location, Span};
+use noirc_frontend::{ast::Visitor, node_interner::NodeInterner, parse_program};
+use num_bigint::BigInt;
+
+use crate::{
+    requests::{to_lsp_location, ProcessRequestCallbackArgs},
+    utils,
+};
+
+pub(super) fn hover_from_visitor(
+    file_id: Option<FileId>,
+    position: Position,
+    args: &ProcessRequestCallbackArgs,
+) -> Option<Hover> {
+    let file_id = file_id?;
+    let file = args.files.get_file(file_id)?;
+    let source = file.source();
+    let (parsed_moduled, _errors) = parse_program(source);
+    let byte_index = utils::position_to_byte_index(args.files, file_id, &position)?;
+
+    let mut finder = HoverFinder::new(args.files, file_id, args.interner, byte_index);
+    parsed_moduled.accept(&mut finder);
```

Comment:
```suggestion
    let (parsed_modules, _errors) = parse_program(source);
    let byte_index = utils::position_to_byte_index(args.files, file_id, &position)?;

    let mut finder = HoverFinder::new(args.files, file_id, args.interner, byte_index);
    parsed_modules.accept(&mut finder);
```

Reply:
Done! (there was another occurrence of this so I changed them both in one go)